compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -32,7 +34,10 @@ pub fn partially_evaluate(
     partial_evaluator.eval()
 }
 
+#[derive(Clone, Debug, Diagnostic, Error)]
 pub enum Error {
+    #[error("partial evaluation error: {0}")]
+    #[diagnostic(code("Qsc.Partial.Eval"))]
```

Comment:
The error code should probably follow the naming convention used for all the other errors: `Qsc.ErrorCategory.ErrorName`
```suggestion
    #[diagnostic(code("Qsc.PartialEval.EvaluationFailed"))]
```